src/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs

```diff
@@ -107,27 +118,64 @@ PruneResult PruneOverpostedData(JsonNode? documentBody, EvaluationResults evalua
         Dictionary<string, string[]> ValidationErrorsFrom(EvaluationResults results)
         {
             var validationErrors = new Dictionary<string, string[]>();
-            var val = new List<string>();
             foreach (var detail in results.Details)
             {
                 var propertyName = string.Empty;
-
                 if (detail.InstanceLocation != null && detail.InstanceLocation.Segments.Length != 0)
                 {
                     propertyName = $"{detail.InstanceLocation.Segments[^1].Value} : ";
                 }
                 if (detail.Errors != null && detail.Errors.Any())
                 {
-                    foreach (var error in detail.Errors)
+                    foreach (var error in detail.Errors.Select(x => x.Value))
                     {
-                        val.Add($"{propertyName}{error.Value}");
-                        validationErrors.Add(propertyName, val.ToArray());
+                        var splitErrors = SplitErrorDetail(error, propertyName);
+
+                        foreach (var splitError in splitErrors)
+                        {
+                            if (validationErrors.ContainsKey(splitError.Key))
+                            {
+                                var existingErrors = validationErrors[splitError.Key].ToList();
+                                existingErrors.AddRange(splitError.Value);
+                                validationErrors[splitError.Key] = existingErrors.ToArray();
+                            }
+                            else
+                            {
+                                validationErrors.Add(splitError.Key, splitError.Value);
+                            }
+                        }
                     }
                 }
             }
             return validationErrors;
         }
     }
+
+    private static readonly Regex _propertyRegex = new Regex("\"([^\"]*)\"", RegexOptions.Compiled);
```

Comment:
Please add a comment that describes the purpose of this regex. Bonus points for a better variable name that describes the purpose.